src/extension.ts

```diff
@@ -212,6 +213,34 @@ export async function activate(context) : Promise<extensionapi.ExtensionAPI> {
     }, this);
     await registerYamlSchemaSupport();
 
+
+      const provider: vscode.TextDocumentContentProvider = <vscode.TextDocumentContentProvider>{
+        onDidChange: null,
+        provideTextDocumentContent: (uri: vscode.Uri, token: vscode.CancellationToken): Thenable<string | undefined> => {
+            if (uri.authority === 'loadkubernetescore') {
+                const value = querystring.parse(uri.query).value;
+                if (!value) {
+                    vscode.window.showErrorMessage(`Missing required value in uri: ${uri}`);
+                    return undefined;
+                }
+                return new Promise((resolve, reject) => {
+
+                    kubectl.invokeWithProgress(" -o json get " + value, `Loading ${value}...`, (result, stdout, stderr) => {
```

Comment:
can we make this async ie `const getData =  await kubectl.invokeAsyncWithProgress(command)`